### ISSUE TITLE
Remove `prefix_is_for_params` from PID_Ros

### ISSUE THREAD

**Issue by bmagyar** [ros-controls/control_toolbox#156]
https://github.com/ros-controls/control_toolbox/pull/129#pullrequestreview-1371661251

**Comment by christophfroehlich** [comment]
quoting @bmagyar 
> yes, I think the idea was to move away from that parameter and maybe even entirely disable that functionality or have one default and not allow different ones

**Comment by destogl** [comment]
I agree on "no string manipulation" as this is the issue here. We were removing "/" from parameters and then the interface names of the following controllers that habe prefix with "/" in it were suddenly invalidated.

If we use the local topic names, it means with the node prefix "~/", do we require parameter prefix? I am not sure if we need this for embedding PID to other controllers

**Comment by ViktorCVS** [comment]
@christophfroehlich Since this issue has been resolved, do you think it’s a good time to make those changes?
https://github.com/ros-controls/ros2_controllers/pull/1585#issuecomment-2985693108

**Comment by christophfroehlich** [comment]
yess, please!
